src/ol_openedx_checkout_external/views.py

```diff
@@ -0,0 +1,96 @@
+"""Views for External Checkout"""
+
+import logging
+
+from common.djangoapps.course_modes.models import CourseMode
+from django.conf import settings
+from django.http import HttpResponseRedirect
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from edx_rest_framework_extensions.auth.session.authentication import (
+    SessionAuthenticationAllowInactiveUser,
+)
+from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+
+log = logging.getLogger(__name__)
+
+
+class ExternalCheckoutView(DeveloperErrorViewMixin, APIView):
+    """
+    An API View for external checkout that redirects to the marketing site for checkout
+
+    **Example Requests**
+
+    GET /checkout-external/?sku=TESTSKU/
+
+    **Example Responses**
+
+    The API will return two types of response codes in general
+    302, 400
+
+    A 302 redirect for the marketing site checkout would be returned in case the request was successful
+
+    A 400 would be returned in case there are errors in the request
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (IsAuthenticated,)
+    http_method_names = ["get"]
+
+    def get(self, request):
+        """
+        External checkout request
+        """
+        if settings.MARKETING_SITE_CHECKOUT_URL is None or "":
```

Comment:
Did you mean to have this raise this error if `MARKETING_SITE_CHECKOUT_URL` is either `None` or `""`? Python is evaluating this as `(settings.MARKETING_SITE_CHECKOUT_URL is None) or ""`, so if  `MARKETING_SITE_CHECKOUT_URL=""`, then that turns into `("" is None) or ""` -> `False or ""` -> `""`, which is falsy.

I think `if not settings.MARKETING_SITE_CHECKOUT_URL` is what you're looking for here.